src/L2DebtManager.sol

```diff
@@ -0,0 +1,183 @@
+        return debtRatioOf(user) > liquidationThreshold;
+    }
+
+    function collateralOf(address user) public view returns (uint256) {
+        return _collaterals[user];
+    }
+
+    function borrowingOf(address user) public view returns (uint256) {
+        return _borrowings[user];
+    }
+
+    /// Debt ratio is calculated as the ratio of the debt to the collateral value in USDC
+    // it returns the ratio in basis points (1e4)
+    function debtRatioOf(address user) public view returns (uint256) {
+        uint256 eEthPriceInUSDC = getEEthPriceInUsdc(); // assuming this returns the price scaled to 6 decimals
+        uint256 debtValue = _borrowings[user];
+        uint256 collateralValue = (_collaterals[user] * eEthPriceInUSDC) / 1e18; // adjust for eETH's 18 decimals
+
+        require(collateralValue > 0, "ZERO_COLLATERAL_VALUE");
+
+        return (debtValue * 1e4) / collateralValue; // result in basis points
+    }
+
+    function remainingBorrowingCapacityInUSDC(address user) public view returns (uint256) {
+        uint256 eEthPriceInUSDC = getEEthPriceInUsdc();
+        uint256 collateralValue = (_collaterals[user] * eEthPriceInUSDC) / 1e18;
+        uint256 maxBorrowingAmount = collateralValue * liquidationThreshold / 1e4;
+        return maxBorrowingAmount > _borrowings[user] ? maxBorrowingAmount - _borrowings[user] : 0;
+    }
+
+    function liquidEEthAmount() public view returns (uint256) {
+        return eETH.balanceOf(address(this)) - totalCollateralAmount;
+    }
+
+    function liquidUsdcAmount() public view returns (uint256) {
+        return USDC.balanceOf(address(this)) - totalBorrowingAmount;
+    }
+
+    // returns eETH price in USDC
+    function getEEthPriceInUsdc() private view returns (uint256) {
+        // TODO: replace it with the actual oracle interaction
+        return eEthPriceInUSDC;
+    }
+
+    function convertEETHtoUSDC(uint256 eethAmount) private view returns (uint256) {
+        uint256 eEthPriceInUSDC = getEEthPriceInUsdc();
+        return eethAmount * eEthPriceInUSDC / 1e18;
+    }
+
+    function getCollateralAmountForDebt(uint256 debtUsdcAmount) public view returns (uint256) {
+        uint256 eEthPriceInUSDC = getEEthPriceInUsdc();
+        return debtUsdcAmount * 1e18 / eEthPriceInUSDC;
+    }
+
+    // Internal functions
+
+    // Use the deposited collateral to pay the debt
+    function _repayWithEEthCollateral(address user, uint256 repayDebtAmount) internal returns (uint256 collateralAmount) {
+        collateralAmount = getCollateralAmountForDebt(repayDebtAmount);
+
+        _borrowings[user] -= repayDebtAmount;
+        _collaterals[user] -= collateralAmount;
+
+        totalBorrowingAmount -= repayDebtAmount;
+        totalCollateralAmount -= collateralAmount;
+        
+        emit RepaidWithEETH(user, repayDebtAmount, collateralAmount);
+    }
+
+    // Setters
+
+    function setLiquidationThreshold(uint256 newThreshold) external {
```

Comment:
For the actual implementation, I think it should be built in a way that allows interaction with different pools in case we need to access more debt. Hence, we will have multiple liquidation thresholds.

Does this equation make sense for a way to aggregate the positions into one liquidation threshold we can apply to our users?

 Weighted Average Calculation
 \[ \text{Liquidation Threshold} = \frac{\sum (\text{Collateral}_i \text{ in ETH} \times \text{Liquidation Threshold}_i)}{\text{Total Collateral in ETH}} \]

Reply:
i like your fancy math. yup, we should consider it :) but good with being focused only on Aave for now for the initial launch.